Upgrade to Core22

nicolasbock opened the pull request.

nicolasbock-canonical committed:
Switch to `core20`

Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -63,14 +56,13 @@ parts:
       - elixir
       - erlang
-    plugin: make
+    plugin: dump
```
Since you're overriding the build step handler completely, it's better to use the "nil" plugin instead.

nicolasbock replied:
Ah cool, will fix.

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -91,7 +83,7 @@ parts:
       CPU_COUNT=$(lscpu --parse=CPU | grep -v "^#" | wc -l)
 
-      export DESTDIR=${SNAPCRAFT_PART_INSTALL}
+      export DESTDIR="$(readlink --canonicalize ${SNAPCRAFT_PART_INSTALL})"
```
Is this necessary? Under normal circumstances the part install directory shouldn't include symlinks.

nicolasbock replied:
Let me revisit that. When I tried that the path (or some component) ended up being relative which confused the `install` binary.

cmatsuoka replied:
It wouldn't hurt, it's just that it's not very common to use it but it will work.

nicolasbock replied:
I agree. I checked this again and as you said the path is absolute.

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -112,7 +104,7 @@ parts:
     after:
       - erlang
-    plugin: make
+    plugin: dump
```
You can use "nil" here as well.

nicolasbock replied:
Thanks

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -124,10 +116,13 @@ parts:
 
       CPU_COUNT=$(lscpu --parse=CPU | grep -v "^#" | wc -l)
 
+      export DESTDIR="$(readlink --canonicalize ${SNAPCRAFT_PART_INSTALL})"
```
About the `CPU_COUNT` variable above, Snapcraft should provide `SNAPCRAFT_PARALLEL_BUILD_COUNT`.

nicolasbock replied:
Ah. I was looking for a variable like that but couldn't find it in the documentation. I'll fix this.

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -136,11 +131,12 @@ parts:
 
   erlang:
-    plugin: autotools
+    plugin: dump
```
Plugin nil should work here too.

nicolasbock replied:
Thanks

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -124,10 +116,13 @@ parts:
       CPU_COUNT=$(lscpu --parse=CPU | grep -v "^#" | wc -l)
 
+      export DESTDIR="$(readlink --canonicalize ${SNAPCRAFT_PART_INSTALL})"
+      export PREFIX=/usr
```
Is this Makefile really accepting `PREFIX` as an environment variable instead of Makefile veriable? ~~In any case, if this is the installation prefix it should be `$SNAPCRAFT_PART_INSTALL/usr` instead of  `/usr`, otherwise files will be installed on the host's system directories.~~ Update: Sorry, overlooked the `DESTDIR` setting above.

nicolasbock replied:
Yes, the `Makefile` uses
```make
DESTDIR ?=
PREFIX ?= /usr/local
```

cmatsuoka commented on snap/snapcraft.yaml:
```diff
@@ -122,14 +112,12 @@ parts:
-      CPU_COUNT=$(lscpu --parse=CPU | grep -v "^#" | wc -l)
-
-      export DESTDIR="$(readlink --canonicalize ${SNAPCRAFT_PART_INSTALL})"
+      export DESTDIR="${SNAPCRAFT_PART_INSTALL}"
```
Please check if this makefile accepts `DESTDIR` as an environment variable since the traditional way to do that is running `make DESTDIR=/foo`.

nicolasbock replied:
And here also
```make
PREFIX ?= /usr/local
```

nicolasbock commented on snap/snapcraft.yaml:
```diff
@@ -162,15 +150,15 @@ parts:
-      make -j ${CPU_COUNT}
-      make install
+      make -j ${SNAPCRAFT_PARALLEL_BUILD_COUNT}
+      echo "[DEBUG] done with building"
```
@cmatsuoka If you look at the CI output then this line is never printed. It is as if the `build` step is not completing.

nicolasbock commented on snap/snapcraft.yaml:
```diff
@@ -162,15 +150,15 @@ parts:
-      make install
+      make -j ${SNAPCRAFT_PARALLEL_BUILD_COUNT}
+      echo "[DEBUG] done with building"
+      exit 1
```
Instead of terminating because of the `exit`, `snapcraft` continues with the next part, see

https://github.com/nicolasbock/rabbitmq-server-snap/runs/7997665722?check_suite_focus=true#step:3:5865

nicolasbock commented on snap/snapcraft.yaml:
```diff
@@ -162,15 +150,15 @@ parts:
 
-      make -j ${CPU_COUNT}
-      make install
+      make -j ${SNAPCRAFT_PARALLEL_BUILD_COUNT}
```
This line is executed, see

https://github.com/nicolasbock/rabbitmq-server-snap/runs/7997665722?check_suite_focus=true#step:3:2769

nicolasbock-canonical committed:
Switch to `autotools` plugin for `erlang`

As suggested by Dani Llewellyn [1].

[1] https://forum.snapcraft.io/t/snapcraft-not-completing-override-build-script/31435/9?u=nicolasbock

Co-Authored-By: Dani Llewellyn <diddledani@ubuntu.com>
Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

nicolasbock-canonical committed:
Switch to `make` plugin for `elixir`

As suggested by Dani Llewellyn [1].

[1] https://forum.snapcraft.io/t/snapcraft-not-completing-override-build-script/31435/10?u=nicolasbock

Co-Authored-By: Dani Llewellyn <diddledani@ubuntu.com>
Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

nicolasbock-canonical committed:
Upgrade to `core22`

Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

nicolasbock-canonical committed:
Enable tests for `elixir`

Signed-off-by: Nicolas Bock <nicolas.bock@canonical.com>

nicolasbock commented:
@diddledani Thanks for your help!
